site/app/models/gradeable/AutoGradedGradeable.php

```diff
@@ -134,7 +134,8 @@ public function getTotalPercent($clamp = false) {
 
     /**
      * Gets the points the submitter earned for the active version
-     * @return int
+     * This should always return an int, except when it returns NAN, which is a float
+     * @return float
```

Comment:
I know this is already merged but shouldn't this have been `int|float` rather than just `float` since it can return either?

Reply:
Yep, good catch.  I'm not sure why the static analysis didn't catch that actually.  Do you want to make a PR to fix it, should I make a PR to fix it, or is it minor enough to leave as-is?

Reply:
I just made a PR to fix it. Simple enough change https://github.com/Submitty/Submitty/pull/7941 if you wouldn't mind reviewing.

Reply:
The SA didn't catch it as it's not technically incorrect. An int can be a float (but not always vice versa), and PHP is happy to typecast an int to a float if we were to add inline type declaration of `: float` to the function.

For example:
```php
<?php

declare(strict_types=1);

function foo($a, $b): float {
  return $a + $b;
}

echo gettype(foo(1, 2)) . "\n";
echo gettype(foo(1, 1.5)) . "\n";
```

both returns will give type `float` here, though if we remove the `: float`, then the first function will return `integer`. 

Given that we're not doing any sort of type analysis on the returned value, the difference between float and integer is unimportant.